### PR TITLE
Issue #11558 Ensure jetty created webapp tmpdir fresh each start

### LINKED ISSUE #11558: New tmp directory should be created on every `WebAppContext` start, if not explicitly configured
**12.0.x**

**ee-9, ee-10**

In previous versions of jetty, if a temporary directory has not been explicitly configured for a `WebAppContext`, each time the webapp starts, jetty will create a fresh temporary directory for it.

In jetty-12, the same temporary directory is being used across webapp restarts.

The culprit is a simple logic bug in `WebAppConfiguration.deconfigure()`:

```java
        Boolean tmpdirConfigured = (Boolean)context.getAttribute(TEMPDIR_CONFIGURED);
        if (tmpdirConfigured != null && !tmpdirConfigured)
            context.setTempDirectory(null);
```

should be:

```java
        Boolean tmpdirConfigured = (Boolean)context.getAttribute(TEMPDIR_CONFIGURED);
        if (tmpdirConfigured == null || !tmpdirConfigured)
            context.setTempDirectory(null);
```

### PR DESCRIPTION
**Pull request by janbartel**
Closes #11558

### CODE REVIEW

File: jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebInfConfiguration.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ public void deconfigure(WebAppContext context) throws Exception
     {
         //if it wasn't explicitly configured by the user, then unset it
         Boolean tmpdirConfigured = (Boolean)context.getAttribute(TEMPDIR_CONFIGURED);
-        if (tmpdirConfigured != null && !tmpdirConfigured)
+        if (tmpdirConfigured == null || !tmpdirConfigured)
             context.setTempDirectory(null);
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebInfConfiguration.java]: I would prefer:
```
        if (!(context.getAttribute(TEMPDIR_CONFIGURED) instanceof Boolean tmpdirConfigured && tmpdirConfigured))
            context.setTempDirectory(null);
```

as this avoids a chance of a caste error, the caste itself, a null check and a variable.

File: jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/WebInfConfiguration.java [hunk 2]
```diff
@@ -85,7 +85,7 @@ public void deconfigure(WebAppContext context) throws Exception
     {
         //if it wasn't explicitly configured by the user, then unset it
         Boolean tmpdirConfigured = (Boolean)context.getAttribute(TEMPDIR_CONFIGURED);
-        if (tmpdirConfigured != null && !tmpdirConfigured)
+        if (tmpdirConfigured == null || !tmpdirConfigured)
```
Comment (gregw) [inline jetty-ee9/jetty-ee9-webapp/src/main/java/org/eclipse/jetty/ee9/webapp/WebInfConfiguration.java]: ditto

**Review by gregw — CHANGES_REQUESTED** [review]
one niggle

File: jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebInfConfiguration.java (hunk 1 above)
Comment (janbartel) [reply jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebInfConfiguration.java]: I think it's more confusing, but ok.

**Comment by lorban** [comment]
Is it possible to write a test?

**Comment by janbartel** [comment]
@lorban  done, please rereview

**Review by lorban — APPROVED** [review]

**Pull request merged**